### ISSUE TITLE
Разбор значений и свойств

### ISSUE THREAD

**Issue by sfaqer** [otymko/gitrules#12]
При разборе правил, каталоги для значений и свойств создаются с именем кода элемента (1, 2, 3... 39 т.д.), хотелось бы видеть в наименовании каталога наименование значения\свойства, на текущий момент визуальный поиск конкретного элемента затруднителен.
![default](https://user-images.githubusercontent.com/32082417/41702203-9cc3f3d4-7572-11e8-8d08-4294e4f3ebfb.png)

**Comment by otymko** [comment]
Надо подумать как сделать. Для значений это можно реализовать. Для свойств уже проблема.

**Comment by otymko** [comment]
@sfaqer Например это может выглядеть так:
![image](https://user-images.githubusercontent.com/11651847/41707137-a5ee9c76-7567-11e8-8b8e-f9e97816a5d1.png)

**Comment by otymko** [comment]
Формат имени:
`"[ПорядковыйНомер] " + НаименованиеЭлемента`

**Comment by otymko** [comment]
@sfaqer 
Можно попробовать: https://github.com/otymko/gitrules/releases/tag/0.2.2
P.S. для установки:
``` opm install -f path/to/file ```

**Comment by otymko** [comment]
Вытекающая проблема -> длина путей увеличена и теперь:
![image](https://user-images.githubusercontent.com/11651847/41755025-726058d4-75ff-11e8-8292-afc51c9dae15.png)

**Comment by sfaqer** [comment]
@otymko 
Столкнулся с проблемой аналигочной 
![](https://user-images.githubusercontent.com/11651847/41755025-726058d4-75ff-11e8-8292-afc51c9dae15.png)

В недрах ишуйтрекера уноскрипта есть даваний ишуй:
https://github.com/EvilBeaver/OneScript/issues/67

Я так понимаю каких то путей обхода нет.
Можно сократить некоторые очевидные наименования (ПравилаКонвертацииОбъектов -> ПКО, ПравилаВыгрузкиДанных->ПВД), так же можно перед сохранением проверять длину пути итоговую, и обрезать в наименовании например приёмник свойства.

**Comment by otymko** [comment]
@sfaqer проблема в том что у меня какой-то ПКО может лежать по следующему путь в конвертации:
Документы -> ЕГАИС -> Версия 1 -> Остатки ЕГАИС и т.п.

Большую часть пусти будут все равно занимать наименования свойств как в моем примере. Поэтому из-за ограничения windows мало что можно сделать -> Наименование группа + Наименования свойств очень длинными получаются.

**Comment by sfaqer** [comment]
@otymko я так понял это не ограничение windows, а ограничения дотнета который внутри уноскрипта, но в нашем случае это ничего не меняет.
В любом случае добавить какую либо проверку на длину пути стоит, и как либо его сократить.
Насколько я понимаю мастер сейчас тоже будет падать, если будет достаточно большая иерархия каталогов в ПКО, даже если свойства\значения будут идентификатором.

**Comment by otymko** [comment]
@sfaqer Да, если иерархия будет больше чем 260 символов то и мастер упадет. Я проекты делаю обычно в 1-2 папках от корня диска.

Надо ждать решения от oscript.

**Comment by bf0rce** [comment]
Проблема длинных путей решается установкой ключа реестра (Только для Windows 10, 1607) и перезагрузкой:
`HKLM\SYSTEM\CurrentControlSet\Control\FileSystem\LongPathsEnabled = 1 (Тип: REG_DWORD)`
https://habr.com/post/307186/

**Comment by otymko** [comment]
@bf0rce обязательно на днях попробую!

**Comment by bf0rce** [comment]
@otymko, из подробностей могу указать следующее.

1. При обращении к файлам и каталогам из под OneScript будет ошибка.
1. Ошибку можно обойти, если создавать каталоги из записывать файлы, например, через comОбъект FileSystemObject. Видимо, там напрямую идет Windows API (или как там это называется).
1. Придется еще сделать настройку `git config --system core.longpaths true`, иначе сам гит тоже может ругаться.
1. При запуске кода внутри платформы правила разберутся нормально без ошибок.

**Comment by ViktorErmakov** [comment]
Добрый день, коллеги!
Скажите, а какие то работы идут по исправлению ситуации с длинной пути 260 символов? Я не могу разложить правила уже месяц))

**Comment by bf0rce** [comment]
На данный момент те способы, о которых я писал выше не работают. Мы свои проблемы пока обходим локальными вызовами команд PowerShell. Он с длинными путями, как оказалось, работает без ошибок.

**Comment by ViktorErmakov** [comment]
Можете дать какую то инструкцию, или в статье по разбору правил выложить к примеру?
Я лично забросил хранить правила в гите с тех пор, т.к. не знаю как решить данную проблему. 
Получается что проект этот для меня немного умер.

**Comment by bf0rce** [comment]
Собственно, инструкции нет. Нас пока спасают просто топорные вставки кода, наподобие такого.
```bsl
	ЗапуститьПриложение("PowerShell.exe -WindowStyle Hidden -Command ""New-Item -ItemType Directory -Force -Path '" + ОбъединитьПути(ТекущийКаталог(), КаталогКоллекции) + "'""",, Истина, КодВозврата);
	Если КодВозврата > 0 Тогда
		ВызватьИсключение("");
	КонецЕсли;
```
Ну а текстовые файлы пишутся через объект `ADODB.Stream`. Он тоже работает без проблем с длинными путями.
